students/MikeShand/lesson03/mailroom.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+
+
+
+donors=[['Andy', [10.00]],['Bill',[15.00, 25.00]],['Chuck',[20.00,30.00,40.00]]]
+
+def donor_list():
+    donor_list=[]
+    for x in donors:
+        print(x[0])
+    thank_you()
+
+
+def thank_you():
+    print('Please enter the donor name\n (Type "list" for a list of current donor names)\n Press "q" to return to console')
+    cd=input(':')
+
+    if cd.lower()=='list':
+        donor_list()
+
+    elif cd.lower()=='q':
+        console()   
+
+    else:
+
+        amount=float(input('Please enter the donation amount:'))
+
+        for j,k in enumerate(donors):
+            if cd==k[0]:
+                donors[j][1].append(amount)
+                break
+        else:
+            donors.append([cd,[amount]])
+
+        print('Thank you {} for your generous donation of ${:.2f}'.format(cd,amount))
+
+def donor_report():
+    print('Here is a list of donors and contributions')
+    print('Name                 Total                Donations            Average')
+    for data in donors:
+
+        print("|{:<20}|{:<20}|{:<20}|{:<20}|".format(data[0],sum(data[1]),len(data[1]),sum(data[1])/len(data[1])))
+        #print(f'{data[0]:<20}'+f'${sum(data[1]):20}'+f'{len(data[1]):20}'+f'${sum(data[1])/len(data[1]):<20}')
+
+    console()

```

Comment:
these calls back to console actually cause a bug.

if you run your program, and select report option (2) couple times then try to quit with option 3 it won't quit.

in your main function `console` your while loop will put the user back into the main prompt naturally, so having additional calls to `console` in other functions causes execution flow to mess up.